### PR TITLE
Adapt testsuites of the  Language server validation and build features for DevWorkspaceHappyPath

### PR DESCRIPTION
**Pull request by musienko-maxim**
### What does this PR do?
We have the depended issue: https://github.com/eclipse/che/issues/20385. We need to adapt the existed DevWorkspaceHappyPath test to this testflaw. Currently it adapts the DevWorkspaceHappyPath and  Language server validation test cases

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/20385.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: tests/e2e/tests/e2e_happy_path/DevWorkspaceHappyPath.spec.ts [hunk 1]
```diff
@@ -10,35 +10,185 @@
...
+        await editor.waitEditorAvailable(classPathFilename);
+        await editor.type(classPathFilename, Key.chord(Key.CONTROL, 'a'), 1);
+        await editor.performKeyCombination(classPathFilename, Key.DELETE);
+        await editor.type(classPathFilename, classpathText, 1);
+        await editor.waitTabWithSavedStatus(classPathFilename);
+    }
+}
+
+// when we use devfile v.2 the test app. is opened in the separate window instead of widget
+async function switchApptWindowAndCheck(contentLocator: By) {
```
Comment (dmytro-ndp) [inline tests/e2e/tests/e2e_happy_path/DevWorkspaceHappyPath.spec.ts]: Looks like a typo - redundant "**t**" in method name "switchApp**t**WindowAndCheck()".

File: tests/e2e/tests/e2e_happy_path/DevWorkspaceHappyPath.spec.ts [hunk 2]
```diff
@@ -10,35 +10,185 @@
...
+
+        test('Codenavigation', async () => {
+            await editor.moveCursorToLineAndChar(javaFileName, 32, 17);
+            await editor.performKeyCombination(javaFileName, Key.chord(Key.CONTROL, Key.F12));
+            await editor.waitEditorAvailable(codeNavigationClassName, TimeoutConstants.TS_EDITOR_TAB_INTERACTION_TIMEOUT * 4);
+        });
 });
+    suite('Validation of workspace build and run', async () => {
+        test('Build application', async () => {
+        const taskName: string = 'build';
```
Comment (dmytro-ndp) [inline tests/e2e/tests/e2e_happy_path/DevWorkspaceHappyPath.spec.ts]: inaccurate line indentation - should be 4 spaces to the right.

**Comment by dmytro-ndp** [comment]
[crw-ci-test]

File: tests/e2e/tests/e2e_happy_path/DevWorkspaceHappyPath.spec.ts [hunk 3]
```diff
@@ -10,35 +10,185 @@
...
+import { Editor } from '../../pageobjects/ide/Editor';
+import { Key, error, By } from 'selenium-webdriver';
+import { DriverHelper } from '../../utils/DriverHelper';
+import { Logger } from '../../utils/Logger';
+import { Terminal } from '../../pageobjects/ide/Terminal';
+import { DialogWindow } from '../../pageobjects/ide/DialogWindow';
+import * as fs from 'fs';
 
-const projectAndFileTests: ProjectAndFileTests = e2eContainer.get(CLASSES.ProjectAndFileTests);
+const ide: Ide = e2eContainer.get(CLASSES.Ide);
```
Comment (sleshchenko) [inline tests/e2e/tests/e2e_happy_path/DevWorkspaceHappyPath.spec.ts:28]: It was supposed to be copying pasting for HappyPath test but it seems to have a lot of differences.
Please, at least separate what is copying/pasted and what is DevWorkspace specific, and where it's appropriate, explain in comments why it's different.

Comment (sleshchenko) [reply tests/e2e/tests/e2e_happy_path/DevWorkspaceHappyPath.spec.ts:28]: like 
```diff
+ await ide.waitNotificationAndClickOnButton('The workspace contains Java projects. Would you like to import them?', 'Yes');
```
which is added for DevWorkspaces only

**Review by dmytro-ndp — APPROVED** [review]

**Review by sleshchenko — APPROVED** [review]

File: tests/e2e/tests/e2e_happy_path/DevWorkspaceHappyPath.spec.ts (hunk 3 above)
Comment (musienko-maxim) [reply tests/e2e/tests/e2e_happy_path/DevWorkspaceHappyPath.spec.ts:28]: done

**Pull request merged**